graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/node/GqlJcrQuery.java

```diff
@@ -281,6 +303,9 @@ private GqlJcrNode getGqlNodeById(String uuid) throws RepositoryException {
     }
 
     private JCRSessionWrapper getSession() throws RepositoryException {
+        if(this.language != null){
+            return getSession(this.language);
+        }
```

Comment:
In general it should  be ok, the only issue I see i that a node fetched with a "localized" JcrQuery will be somewhat different than a node fetched with an classical JcrQuery : i18n properties will be readable without having to set the language on the property field . This may be seen as an improvement (that can be discussed, as it will change the api behaviour just by adding a locale at /jcr level), but would potentially break the cache as these 2 nodes (i18n/non i18n) will share the same cache key. That's the only issue I can see, maybe it will introduce other problems. 
Ideally we should add the locale only for the visibility check, but still use the node without locale ...